pkg/storage/replica.go

```diff
@@ -2746,6 +2746,7 @@ func (r *Replica) applyTimestampCache(
 						txn := ba.Txn.Clone()
 						bumped = txn.Timestamp.Forward(nextTS) || bumped
 						ba.Txn = &txn
+						ba.Timestamp = txn.Timestamp
```

Comment:
please add a comment at this change and below, reiterating that we want to write at the bumped ts.